packages/api/cms-api/src/page-tree/page-tree-read-api.ts

```
@@ -422,6 +420,19 @@ export function createReadApi(
     };
 }
 
+export function paginateNodes<T>(nodes: T[], options: { offset?: number; limit?: number }) {
+    const offset = options.offset ?? -1;
+    const limit = options.limit ?? -1;
+
+    if (offset < 0 || limit <= 0) {
+        return nodes;
```

Should we throw an error here instead?